code/modules/psionics/psion.dm

```diff
@@ -78,22 +82,68 @@
 		if(owner.stats.getPerk(PERK_PSI_PSYCHOLOGIST))
 			psi_max_bonus += 5
 
+		psi_max_bonus += psionic_equipment_check("psi_max_bonus")
+
 		max_psi_points = round(clamp((owner.stats.getStat(STAT_COG) * 0.1), 1, 30)) + psi_max_bonus
 
-		cognitive_potential = round(clamp((owner.stats.getStat(STAT_COG) * 0.5), 0, 3))
+		cognitive_potential = round(clamp((owner.stats.getStat(STAT_COG) * 0.1), 0, cognitive_potential_max), 0.1)
 
+		var/regen_points_timer = (5 MINUTES - cognitive_potential MINUTES)
+		var/min_timer = 1 MINUTES
```

Comment:
Min timer doesn't change so you can probably make it a var owned by the organ instead of initializing it every function call